contracts/context-config/src/mutate.rs

```diff
@@ -360,42 +390,47 @@ impl ContextConfigs {
     pub fn proxy_deployment_callback(
         &mut self,
         #[callback_result] call_result: Result<(), PromiseError>,
-    ) {
-        if let Err(e) = call_result {
-            panic!("Failed to deploy proxy contract: {:?}", e);
+    ) -> PromiseOrValue<()> {
+        if let Ok(_) = call_result {
+            // Calculate actual storage used and refund excess
+            let actual_storage_used = env::storage_usage();
+            let actual_cost =
+                (actual_storage_used as u128) * env::storage_byte_cost().as_yoctonear();
+            let deposit_used = env::attached_deposit().as_yoctonear();
```

Comment:
use `NearToken`, we shouldn't need to convert to raw yocto

https://docs.rs/near-sdk/latest/near_sdk/struct.NearToken.html#method.saturating_sub